common/src/main/java/io/pravega/common/util/ReusableFutureLatch.java

```diff
@@ -202,7 +202,21 @@ public void releaseExceptionallyAndReset(Throwable e) {
             }
         }
     }
-    
+
+    /**
+     * Returns whether or not release has been called.
+     *
+     * @return True if the latch is set to release state.
+     */
+    public boolean isReleased() {
+        boolean state;
+        synchronized (lock) {
+            state = released;
```

Comment:
You can just return from the `synchronized` block. No need for the extra variable.